packages/core-mobile/app/services/earn/EarnService.ts

```
@@ -403,10 +403,15 @@ class EarnService {
       return currentNetworkTransactions
         .concat(oppositeNetworkTransactions)
         .map(transaction => {
+          // find account that matches the transaction's index
+          const account = accountsArray.find(
+            acc => acc.index === transaction.index
+          )
```

should we throw error here if account if not found with the transaction index?